onedal/objective_function/objective_function.py

```diff
@@ -0,0 +1,314 @@
+        y,
+        sample_weight=None,
+        l2_reg_strength=0.0,
+        n_threads=1,
+        raw_prediction=None,
+        queue=None
+    ):
+        if (sample_weight is not None):
+            if (not np.array_equal(sample_weight, np.ones_like(sample_weight))):
+                raise Exception("sample_weigth parameter is not supported")
+        if (n_threads != 1):
+            raise Exception("multithreading is not supported")
+        if (raw_prediction is not None):
+            raise Exception("raw_prediction parameter is not supported")
+
+        value = super()._compute(X, y, coef, "value", 0.0,
+                                 self.fit_intercept, queue)["value"]
+        if l2_reg_strength > 0:
+            value += self.__calculate_regularization(coef, l2_reg_strength)
+        return value
+
+    def loss_gradient(
+        self,
+        coef,
+        X,
+        y,
+        sample_weight=None,
+        l2_reg_strength=0.0,
+        n_threads=1,
+        raw_prediction=None,
+        queue=None
+    ):
+        if (sample_weight is not None):
+            if (not np.array_equal(sample_weight, np.ones_like(sample_weight))):
+                raise Exception("sample_weigth parameter is not supported")
+        if (n_threads != 1):
+            raise Exception("multithreading is not supported")
+        if (raw_prediction is not None):
+            raise Exception("raw_prediction parameter is not supported")
+
+        res = super()._compute(
+            X, y, coef, ["value", "gradient"], 0.0, self.fit_intercept, queue
+        )
+        value, grad = res["value"], res["gradient"]
+        if l2_reg_strength > 0:
+            value += self.__calculate_regularization(coef, l2_reg_strength)
+        return (value, self.change_gradient_format(grad, coef, l2_reg_strength))
+
+    def gradient(
+        self,
+        coef,
+        X,
+        y,
+        sample_weight=None,
+        l2_reg_strength=0.0,
+        n_threads=1,
+        raw_prediction=None,
+        queue=None
+    ):
+        if (sample_weight is not None):
+            if (not np.array_equal(sample_weight, np.ones_like(sample_weight))):
+                raise Exception("sample_weigth parameter is not supported")
+        if (n_threads != 1):
+            raise Exception("multithreading is not supported")
+        if (raw_prediction is not None):
+            raise Exception("raw_prediction parameter is not supported")
+
+        grad = super()._compute(X, y, coef, "gradient", 0.0, self.fit_intercept, queue)[
+            "gradient"
+        ]
+        return self.change_gradient_format(grad, coef, l2_reg_strength)
+
+    def gradient_hessian(
+        self,
+        coef,
+        X,
+        y,
+        sample_weight=None,
+        l2_reg_strength=0.0,
+        n_threads=1,
+        raw_prediction=None,
+        queue=None
+    ):
+        if (sample_weight is not None):
+            if (not np.array_equal(sample_weight, np.ones_like(sample_weight))):
```

Comment:
You are allocating extra array just for check? You should just check elements of sample_weight